Pipfile

```diff
@@ -0,0 +1,29 @@
+[[source]]
+name = "pypi"
+url = "https://pypi.org/simple"
+verify_ssl = true
+
+[packages]
+boto3 = "1.28.62"
+cryptography = "3.3.2"
+deprecated = "1.2.14"
+keyring = "23.4.1"
+pandas = "2.0.3"
```

Comment:
There's some packages we don't require for users: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/2747b12a408729c8496fbf3fdf5d7411f8a8bba6/setup.cfg#L87-L94. but these are all relevant for development.

We can certainly lock the version down or use the `~=` function to be a bit more permissive.  Thoughts?

Reply:
I'll move those packages over.

For versioning IMO I didn't like giving ranges that much when dealing with lock files - Since regenerating the lock file is going to update many dependencies that might not have been intentionally changed. Though I am ok with whichever direction.

Reply:
Could we follow something like this where the pipfile pulls from a setup.cfg and/or pyproject.toml: https://github.com/Sage-Bionetworks-Workflows/py-orca/blob/e3ddde039e2776c6b8f053f7c502fc05dab6f9cd/Pipfile#L6-L13.  One concern I have is that the python community is moving away from setup.cfg as stated in pep-621: https://peps.python.org/pep-0621/.

I agree we shouldn't do ranges, but lets do something with this: https://peps.python.org/pep-0440/#compatible-release

Reply:
Great idea. I will explore this and add a new commit with the changes.

Reply:
I modified the pipfile to pull the dependencies from the `setup.cfg` like `py-orca` is doing - Is this what you had in mind?
https://github.com/Sage-Bionetworks/synapsePythonClient/pull/984/commits/fc0f18d43b840cb9cb7b2d6ad355b1df2d549106